R-package/.Rbuildignore

```diff
@@ -12,3 +12,14 @@
 
 # Code copied in at build time
 ^src/CMakeLists.txt$
+
+# unnecessary files from submodules
+^src/compute/.appveyor.yml$
+^src/compute/.coveralls.yml$
+^src/compute/.travis.yml$
+^src/compute/test/$
+^src/compute/index.html$
+^src/compute/.git$
+^src/compute/.gitignore$
+^src/compute/CONTRIBUTING.md$
+^src/compute/README.md$
```

Comment:
`src/compute` is the boost stuff we pull in as a git submodule. It's important that the random other stuff from that repo isn't bundled with the R package, as `R CMD check` will complain about things like a `.git/` folder being put into a source distribution.